src/macaron/slsa_analyzer/checks/provenance_l3_check.py

```diff
@@ -287,12 +344,34 @@ def run_check(self, ctx: AnalyzeContext, check_result: CheckResult) -> CheckResu
                 check_result["justification"].append("Could not verify level 3 provenance.")
                 return CheckResultType.FAILED
 
+        result_value = CheckResultType.FAILED
         if all_feedback:
-            check_result["justification"].append(
-                "Successfully verified level 3 provenance for the following artifacts",
-            )
-            check_result["justification"].extend(all_feedback)
-            return CheckResultType.PASSED
+            all_results = [result for _, _, result in all_feedback]
+            failed = [
+                result
+                for ci_name, prov_url, result in all_feedback
+                if result.result == _VerifyArtefactResultType.FAILED
+            ]
+            passed = [
+                result
+                for ci_name, prov_url, result in all_feedback
+                if result.result == _VerifyArtefactResultType.PASSED
+            ]
+            skipped = [
+                result for ci_name, prov_url, result in all_feedback if result not in passed and result not in failed
+            ]
+
+            if failed or skipped:
+                result_value = CheckResultType.FAILED
+
+            if passed:
+                check_result["justification"].append("Successfully verified level 3: ")
+            else:
+                check_result["justification"].append("Failed verification for level 3: ")
```

Comment:
I think `result_value = CheckResultType.PASSED` is missing when verification is successful?

Also, if `passed` is not empty, this would override the results even if `failed` or `skipped` are not empty.

```suggestion
            if failed or skipped:
                result_value = CheckResultType.FAILED
                check_result["justification"].append("Failed verification for level 3: ")
            else:
                result_value = CheckResultType.PASSED
                check_result["justification"].append("Successfully verified level 3: ")
```